components/raftstore/src/store/peer.rs

```diff
@@ -1608,6 +1629,40 @@ where
         false
     }
 
+    pub fn maybe_force_forward_commit_index(&mut self) {
+        let expected_alive_voter = match &self.force_leader {
+            Some(ForceLeaderState::ForceLeader {
+                expected_alive_voter,
+            }) => expected_alive_voter,
+            _ => unreachable!(),
+        };
+
+        let mut replicated_idx = self.raft_group.raft.raft_log.persisted;
+        for (peer_id, p) in self.raft_group.raft.prs().iter() {
+            let store_id = self
+                .region()
+                .get_peers()
+                .iter()
+                .find(|p| p.get_id() == *peer_id)
+                .unwrap()
+                .get_store_id();
+            if !expected_alive_voter.contains(&store_id) {
+                continue;
+            }
+            if replicated_idx > p.matched {
+                replicated_idx = p.matched;
+            }
+        }
+
+        if self.raft_group.store().term(replicated_idx).unwrap() < self.term() {
```

Comment:
Should use `map_or` in case replicated_idx is 0 or out of bound. A case is missing.